tutorials/notebook.md

```diff
@@ -34,529 +36,451 @@ npm init -y
+nearly any type.  We will storing our notes in an array of strings at 'notebook/notes'.
+
+``` javascript
+async function addNote(note) {
+
+    let { tree } = await readIdentifierFile(); // Using our new function to retrieve
+
+    const resp = await tree.resolveData(CHAIN_TREE_NOTE_PATH);
+    let notes = resp.value,
+        noteWithTs = addTimestamp(note); // Add a time and date to our new entry
+
+    if (notes instanceof Array) {
+        notes.push(noteWithTs);
+    } else {
+        notes = [noteWithTs];
+    }
 
-In file `notebook/index.js`:
-```javascript
-function addNote(creds, note) {
-  if (!idFileExists()) {
-    console.log("Error: you must register before you can record notes.");
-
-    return;
-  }
-
-  let client = connect(creds), // <--- connect the client
-      identifiers = readIdentifierFile();
-
-  client.resolveData(identifiers.chainId, CHAIN_TREE_NOTE_PATH)
-    .then(function(resp) {
-    // We'll fill this in later.
-  }, function(err) {
-      console.log('Error reading notes: ' + err);
-  });
 }
 ```
 
-### Adding New Notes
-Now we're ready to append our new note to the previously stored notes and write
-that data back to the chain tree. But first, we need a way to add a timestamp to
-the note we're recording. Let's write a small function to add the current
-timestamp to our note.
+You will note that along with our text values we have decided to add a timestamp.  
+This will provide additional immutable information to our notebook to be signed by
+the Tupelo TestNet.  Our application will append the current date and time when each of
+our notes was entered.
 
-In file `notebook/index.js`:
-```javascript
+We will need to add a simple function to support that timestamping.
+
+``` javascript
 function addTimestamp(note) {
-  let ts = new Date().getTime().toString();
-  return ts + '::' + note;
+    let ts = new Date().getTime().toString();
+    return ts + '::' + note;
 }
 ```
 
-Now let's use this function to add to our `addNote()` function to append the new
-note (with timesamp) to the previously stored notes. First, we'll need to make
-sure that the data we read from the chain tree is an array and append the new
-note. If it isn't an array, then the data is either corrupted or empty, so we
-can just discard it for now. We'd add more robust error handling for a
-production app though. Lastly, we'll add the newly appended note data back to
-the chain tree using the `setData()` Tupelo API method.
+The last step we need to take in our addNote function is to actually submit the
+information to be signed!  So we will add the SDK calls to the end of addNote()
+to do just that.  The playTransactions call takes the tree we are changing
+and the change we want to make to the data as arguments.  
 
-In file `notebook/index.js`:
-```javascript
-function addNote(creds, note) {
-  if (!idFileExists()) {
-    console.log("Error: you must register before you can record notes.");
+``` javascript
+    ...
```

Comment:
I understand what you mean with the ellipsis here, but since this in a JS syntax block, it might be better to do something like this instead:

``` javascript
    // inside the addNote function
```

Reply:
Ended up with:
    }
  // ... additional body of addNote ...  
}